tests/test_authority.py

```diff
@@ -123,3 +128,81 @@ class MockResponse(object):
         finally:  # MUST NOT let the previous test changes affect other test cases
             Authority._domains_without_user_realm_discovery = set([])
 
+
+_mocked_tenant_discovery_response = {
+    "authorization_endpoint": "https://contoso.com/placeholder",
+    "token_endpoint": "https://contoso.com/placeholder",
+    }
+
+@patch.multiple(
+    "msal.authority",
+    instance_discovery=DEFAULT,
+    tenant_discovery=Mock(return_value=_mocked_tenant_discovery_response),
+    )
+class TestMsalAuthorityBehaviorsWithoutKnownToDeveloperAuthorities(unittest.TestCase):
+
+    def test_a_known_to_microsoft_authority_should_skip_instance_discovery(self, **mocks):
+        msal.PublicClientApplication("id", authority="https://login.microsoftonline.com/common")
+        mocks["instance_discovery"].assert_not_called()
+
+    def test_validate_authority_boolean_should_skip_instance_discovery(self, **mocks):
+        """Pending deprecation, but kept for backward compatibility, for now"""
+        msal.PublicClientApplication(
+            "id", authority="https://contoso.com/common", validate_authority=False)
+        mocks["instance_discovery"].assert_not_called()
```

Comment:
As above, instance discovery to figure out aliases -  for SSO islands problem.

Reply:
This `validate_authority=False` case is different than the case you commented above. I thought we discussed this in the design doc, that for `validate_authority=False` authority, the authority aliases do not apply, so they can be skipped. Please ping me in Teams so that we can discuss.

Reply:
Test cases are updated to better show the intention. I still think the instance metadata behavior should be SKIPPED in this case. If you still think otherwise, let's discuss offline.